application/src/main/java/uk/nhs/digital/uec/api/service/PostcodeMappingServiceImpl.java

```diff
@@ -23,47 +29,52 @@
 
   @Override
   public List<PostcodeMapping> getByPostCodes(List<String> postCodes)
-      throws InvalidPostcodeException {
+      throws InvalidPostcodeException, NotFoundException {
     List<String> validPostcodes = validatePostCodes(postCodes);
-    List<PostcodeMapping> results = new ArrayList<PostcodeMapping>();
-    validPostcodes.stream().forEach(postcode -> results.addAll(getByPostcode(postcode)));
-    return results;
+    List<PostcodeMapping> location =
+        validPostcodes.stream()
+            .map(this::getByPostcode)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    return validateAndReturn(location);
   }
 
   @Override
-  public List<PostcodeMapping> getByName(String name) {
-    return postcodeMappingRepository.findByName(name).stream()
-        .filter(Optional::isPresent)
-        .map(Optional::get)
-        .collect(Collectors.toList());
+  public List<PostcodeMapping> getByName(String name)
+      throws InvalidParameterException, NotFoundException {
+    List<PostcodeMapping> location =
+        postcodeMappingRepository.findByName(name).stream()
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    return validateAndReturn(location);
   }
 
   @Override
   public List<PostcodeMapping> getByPostCodesAndName(List<String> postCodes, String name)
-      throws InvalidPostcodeException {
+      throws InvalidPostcodeException, NotFoundException {
+    if (CollectionUtils.isEmpty(postCodes) && StringUtils.isAllBlank(name)) {
```

Comment:
I dont know if the && is correct. What happens if the postcode list is empty but the name is not.
We also want to move this into the validate methods, so that we know all the validation is in one place. If we look through the code we are validating here and throwing an exception, but then we are validating again in the validation methods and throwing a different exception and it gets confusing and hard to read.
@jope1 This is a good example of why I was advocating a ValidatorService

Reply:
We agreed we'd raise a ticket for the validator service - SFD-4106 - so we'll pick that up there. (Probably straight after this one actually looking at the remaining tickets in the sprint.) Without a validator service, validation should be performed at the earliest possible opportunity, so that's in the controller for now.

Reply:
changes done